File: 004/004.c
#include <stdio.h>
#include <string.h>

int is_palindrome(long int input)
{
    char a[10] = {'\0'};
    int i, j, r = 1;

    for (i = 0, j = input; j > 0; j /= 10)
        a[i++] = (j % 10) + '0';

    for (i = strlen(a) - 1; i > 0;)
        r &= a[i--] == a[j++];
    
    return r;
}

long int find_largest()
{
    long int a,b,t,m = 0;

    for (a = 999; a > 100; --a) {
        for (b = a; b > 100; --b) {
            t = a * b;
            if (is_palindrome(t) && t > m)
                m = t;
        }
    }

    return m;
}

int main(void)
{
    printf("Result %ld\n", find_largest());
    return 0;
}

File: 004/004.py
def is_palindrome(input):
    z = str(input)
    return z == z[::-1]

def find_largest():
    a,b,m = 0,0,0
    for i in xrange(999, 100, -1):
        for j in xrange(i, 100, -1):
            x = i * j
            if is_palindrome(x) and x > m:
                a,b,m = i,j,x
    return (a, b, m)

print find_largest()

